qcodes/instrument_drivers/Keysight/Keysight_E8267D.py

```diff
@@ -55,10 +60,20 @@ def __init__(self, name, address, step_attenuator=False, **kwargs):
                            get_cmd=':OUTP?',
                            set_cmd='OUTP {}',
                            get_parser=parse_on_off,
-                           # Only listed most common spellings idealy want a
-                           # .upper val for Enum or string
-                           vals=vals.Enum('on', 'On', 'ON',
-                                          'off', 'Off', 'OFF'))
+                           vals=on_off_validator)
+        self.add_parameter('IQmodulator',
+                           get_cmd='DM:STATe?',
+                           set_cmd='DM:STATe {}',
+                           get_parser=parse_on_off,
+                           vals=on_off_validator,
+                           docstring='Enables or disables the internal I/Q modulator. Source can be external or internal.')
+
+        for source in [1, 2]:
+            self.add_parameter(f'IQsource{source}',
+                               get_cmd=f'DM:SOUR{source}?',
+                               set_cmd=f'DM:SOUR{source} {{}}',
+                               vals=vals.Enum('OFF', 'EXT', 'EXT600', 'INT'),
```

Comment:
does it make sense to give these values human-friendly names, and use val_mapping to map them to these SCPI values? i'd really recommend this.

Reply:
From the manual:

*EXTernal*  This choice selects an external 50 ohm source as the I/Q input to I/Q modulator.
*INTernal*    This  choice  is  for  backward  compatibility  with  ESG  E44xxB  models  and  performs  the  same  function  as  the  BBG1  selection.
*BBG1*    This  choice  selects  the  baseband  generator  as  the  source  for  the  I/Q  modulator.
*EXT600*    This  choice  selects  a  600  ohm  impedance  for  the  I  and  Q  input  connectors  and  routes  the  applied  signals  to  the  I/Q  modulator.
*OFF*    This  choice  disables  the  I/Q  inpu
```
What would good names be? 

@astafan8

Reply:
well, i would do smth like the following:
```
{
"external_50_ohm": "EXT",
"internal": "INT",
"baseband_generator": "BBG1",
"external_600_ohm": "EXT600",
"off": "OFF"
}
```
and also add a `docstring=....` kwarg to this `add_parameter` call with the text that explain all these values so that the user does not need to look into the manual.

But if you think it's overkill, or unnecessary humanization of the values, feel free to use whatever you think is reasonable.